src/Aspire.Hosting.Azure/Extensions/AzureEventHubsExtensions.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Azure.Provisioning;
+using Azure.Provisioning.Authorization;
+using Azure.Provisioning.EventHubs;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding the Azure Service Bus resources to the application model.
+/// </summary>
+public static class AzureEventHubsExtensions
+{
+    /// <summary>
+    /// Adds an Azure Event Hubs Namespace resource to the application model. This resource can be used to create Event Hub resources.
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<AzureEventHubsResource> AddAzureEventHubs(
+        this IDistributedApplicationBuilder builder, string name)
+    {
+#pragma warning disable CA2252 // This API requires opting into preview features
+        return builder.AddAzureEventHubs(name, null);
+#pragma warning restore CA2252 // This API requires opting into preview features
+    }
+
+    /// <summary>
+    /// Adds an Azure Event Hubs Namespace resource to the application model. This resource can be used to create Event Hub resources.
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="configureResource">Optional callback to configure the Event Hubs namespace.</param>
+    /// <returns></returns>
+    [RequiresPreviewFeatures]
+    public static IResourceBuilder<AzureEventHubsResource> AddAzureEventHubs(this IDistributedApplicationBuilder builder, string name,
+        Action<IResourceBuilder<AzureEventHubsResource>, ResourceModuleConstruct, EventHubsNamespace>? configureResource)
+    {
+        var configureConstruct = (ResourceModuleConstruct construct) =>
+        {
+            var eventHubsNamespace = new EventHubsNamespace(construct, name: name);
+
+            eventHubsNamespace.Properties.Tags["aspire-resource-name"] = construct.Resource.Name;
+
+            eventHubsNamespace.AssignProperty(p => p.Sku.Name, new Parameter("sku", defaultValue: "Standard"));
+
+            var serviceBusDataOwnerRole = eventHubsNamespace.AssignRole(RoleDefinition.EventHubsDataOwner);
+            serviceBusDataOwnerRole.AssignProperty(p => p.PrincipalType, construct.PrincipalTypeParameter);
+
+            eventHubsNamespace.AddOutput("serviceBusEndpoint", sa => sa.ServiceBusEndpoint);
```

Comment:
@JoshLove-msft:   Is the CDK sending back "ServiceBusEndpoint" for an Event Hubs resource?   This looks wrong.

Reply:
That's the name of the property on the ARM API AFAIK

Reply:
I'm not the only one guilty of this :) 

![image](https://github.com/dotnet/aspire/assets/1844001/948c54d5-edfa-4dc3-b1f9-8205f25f4524)

Reply:
I've updated all the Bicep references and any of my code to use eventHubsEndpoint instead of serviceBusEndpoint, even though that feels weird to me, perversely. But we have to start somewhere :)

Reply:
Looks like you're right.  _([example](https://github.com/Azure/azure-rest-api-specs/blob/969e0846e56a0869203bcc52773415c71115f59e/specification/eventhub/resource-manager/Microsoft.EventHub/stable/2021-11-01/examples/NameSpaces/EHNameSpaceCreate.json#L86))_   
That's an odd choice, but I agree we should keep it aligned with ARM.

Reply:
So @davidfowl -- do I revert back to ServiceBusEndpoint/serviceBusEndpoit for outputs, vars etc? You seem to indicate the opposite with the thumbs up on my change below :)  -- @jsquire you agree with who? I am confused lol

Reply:
@oising:  Sorry, our comments cross-threaded, David's comment was the last visible one and the one that I was responding to.   I defer to you and David on the path forward, but for my two cents, I like your updates.  😄